### DISCUSSION TITLE
Minio docker image not working with HTTPS

### DISCUSSION THREAD

**Discussion by sonicaj** [minio/minio#12845] [Q&A] [answered]
## Expected Behavior

I was using docker image `minio/minio:RELEASE.2021-07-30T00-02-00Z` in a kubernetes cluster. Everything works nicely until i configure a certificate. I am not using the official minio operator but a simple helm chart ( https://github.com/truenas/charts/tree/master/charts/minio/1.3.7 ) which deploys the minio image and we have a node port service to access the minio UI.
Minio console in my example is running on port 9001 and minio API on 9000.
When a certificate is configured, i was expecting to be able to access the minio console UI at https://NODEIP:9001.

## Current Behavior

Currently when i access https://NODEIP:9001, i am presented with a blank screen for some time and then after some time it takes me directly to the graphical interface without logging me in. In some retries i also found it presented me with the login page but login kept on failing. Please see below screenshot for the errors being logged in the pod.

<img width="1400" alt="Screenshot 2021-07-31 at 10 28 06 PM" src="https://user-images.githubusercontent.com/17968138/127747895-1c5d6683-b659-4427-8351-100d78d4bd10.png">

I tried setting `MINIO_SERVER_URL` to `https://NODEIP:9001`, `https://NODEIP`, `https://NODEDOMAIN:9001` and `https://NODEDOMAIN` but that did not help improve the situation.

## Steps to Reproduce (for bugs)

1. Install the k8s resources from here ( https://gist.github.com/sonicaj/e7dd35b2bc2097bf615c7ee9068aab2c )
2. Please change nodeip/nodedomain in the deployment to reflect yours
3. Try to access the minio cosole with node port service running at 9001
4. We can see the errors being generated in the logs of minio pod

## Context

I am trying to run minio docker image in kubernetes with TLS. It was working nicely before console changes were introduced ( the TLS part ).

## Your Environment

* Version used (`minio --version`): `minio/minio:RELEASE.2021-07-30T00-02-00Z`
* Server setup and configuration: Kubernetes cluster - 1.21 version

**Comment by harshavardhana** [comment, marked as answer]
You are missing couple of environment variables https://github.com/minio/minio#test-minio-connectivity

**Reply by sonicaj** [reply]
@harshavardhana i tried setting `MINIO_BROWSER_REDIRECT_URL ` and that has not helped either. Can you please clarify if you are referring to some other environment variable as well ? ( `MINIO_SERVER_URL ` was already set )

**Reply by sonicaj** [reply]
Also if i am not using a certificate in the gist shared, without setting both `MINIO_BROWSER_REDIRECT_URL` and `MINIO_SERVER_URL` - minio works really nicely. I suspect it has something to do with TLS configuration ?

**Reply by harshavardhana** [reply]
> Also if i am not using a certificate in the gist shared, without setting both `MINIO_BROWSER_REDIRECT_URL` and `MINIO_SERVER_URL` - minio works really nicely. I suspect it has something to do with TLS configuration ?

Well it is because your TLS doesn't have the right certs for the endpoint? 

MINIO_SERVER_URL needs to point to port 9000
MINIO_BROWSER_REDIRECT_URL needs to point to port 9001

If you make sure these are set it will work fine.

**Reply by sonicaj** [reply]
Thank you for clarifying @harshavardhana, it indeed works if i use the domain name with the ports you specified.
However if I use the node ip which is also present in the SAN extension of the certificate, it doesn't work with the ports specified and i see similar errors logged. I understand it is unconventional using ip addresses in certificates but was curious that is this expected behavior ?
To clarify, if an IP Address is specified in the certificate and minio is to consume that cert, we are seeing the same login errors being logged and it not allowing us to log in.
```
        X509v3 extensions:
            X509v3 Subject Alternative Name:
                IP Address:192.168.0.3, DNS:domainwhichworked.com
```

**Reply by harshavardhana** [reply]
> Thank you for clarifying @harshavardhana, it indeed works if i use the domain name with the ports you specified.
> However if I use the node ip which is also present in the SAN extension of the certificate, it doesn't work with the ports specified and i see similar errors logged. I understand it is unconventional using ip addresses in certificates but was curious that is this expected behavior ?
> To clarify, if an IP Address is specified in the certificate and minio is to consume that cert, we are seeing the same login errors being logged and it not allowing us to log in.
> 
> ```
>         X509v3 extensions:
>             X509v3 Subject Alternative Name:
>                 IP Address:192.168.0.3, DNS:domainwhichworked.com
> ```

You should just use the domain name and not overly complicate things. Use TLS that works with the domain name that's it.
